### ISSUE TITLE
Microsecond precision is lost on shift operations

### ISSUE THREAD

**Issue by alvinlindstam** [bitwalker/timex#352]
### Steps to reproduce

```
iex(228)> n = Timex.now()                                                  
#<DateTime(2017-08-04T16:44:51.568241Z Etc/UTC)>
iex(229)> {microseconds, _precision } = n.microsecond                      
{568241, 6}
iex(230)> Timex.shift(n, microseconds: -microseconds).microsecond          
{0, 0}
iex(231)> Timex.shift(n, microseconds: -rem(microseconds, 100)).microsecond
{568200, 4}
```
### Description of issue
When a datetime is shifted, the microseconds seem to reduce the precision to whatever could be assumed based on the number of trailing zeroes.

I would expect all of the datetimes above to keep precision at 6.

I don't really know what I'd expect if the precision is low on the input DateTime, and single digit microseconds are shifted.

**Comment by bitwalker** [comment]
Yeah we definitely should not be reducing precision when we were given an input with a precision already set.

**Comment by novaugust** [comment]
This bug is still around. The PR to fix it actually only fixed the implementation for `%NaiveDateTime{}`

Here's a repro on 3.3
```
iex(2)> n = Timex.now()
#DateTime<2019-04-22 15:55:59.005002Z>
iex(3)> {microseconds, _precision } = n.microsecond
{5002, 6}
iex(4)> Timex.shift(n, microseconds: -microseconds).microsecond
{0, 0}
iex(5)> Timex.shift(n, microseconds: -rem(microseconds, 100)).microsecond
{5000, 3}
```

**Comment by novaugust** [comment]
Buuut it's fixed on 3.5! So nevermind and thanks =)
